packages/test-base/src/androidTest/kotlin/io/realm/kotlin/test/shared/MutableRealmTests.kt

```diff
@@ -675,4 +680,98 @@ class MutableRealmTests {
             }
         }
     }
+
+    @Test
+    fun deleteAll() {
+        realm.writeBlocking {
+            for (i in 0..9) {
+                copyToRealm(Sample())
+                copyToRealm(SampleWithPrimaryKey().apply { primaryKey = i.toLong() })
+                copyToRealm(
+                    EmbeddedParent().apply {
+                        id = "level$i-parent"
+                        child = EmbeddedChild().apply {
+                            id = "level$i-child1"
+                        }
+                    }
+                )
+            }
+            assertEquals(10, query<Sample>().count().find())
+            assertEquals(10, query<SampleWithPrimaryKey>().count().find())
+            assertEquals(10, query<EmbeddedParent>().count().find())
+            assertEquals(10, query<EmbeddedChild>().count().find())
+            deleteAll()
+            assertEquals(0, query<Sample>().count().find())
+            assertEquals(0, query<SampleWithPrimaryKey>().count().find())
+            assertEquals(0, query<EmbeddedParent>().count().find())
+            assertEquals(0, query<EmbeddedChild>().count().find())
+        }
+    }
+
+    private fun addSampleData(realm: Realm) {
+        realm.writeBlocking {
+            for (i in 0..9) {
+                copyToRealm(Sample())
+                copyToRealm(SampleWithPrimaryKey().apply { primaryKey = i.toLong() })
+                copyToRealm(
+                    EmbeddedParent().apply {
+                        id = "level$i-parent"
+                        child = EmbeddedChild().apply {
+                            id = "level$i-child1"
+                        }
+                    }
+                )
+            }
+        }
+    }
+
+    @Test
+    fun deleteAll_onlyDeletesCurrentSchema() {
+        addSampleData(realm)
+        realm.writeBlocking {
+            assertEquals(10, query<Sample>().count().find())
+            assertEquals(10, query<SampleWithPrimaryKey>().count().find())
+            assertEquals(10, query<EmbeddedParent>().count().find())
+            assertEquals(10, query<EmbeddedChild>().count().find())
+        }
+        val realm2 = Realm.open(RealmConfiguration.Builder(schema = setOf(Sample::class)).directory(tmpDir).build())
+        realm2.writeBlocking {
+            assertEquals(10, query<Sample>().count().find())
+            deleteAll()
+            assertEquals(0, query<Sample>().count().find())
+        }
+        realm.writeBlocking {
+            assertEquals(0, query<Sample>().count().find())
+            assertEquals(10, query<SampleWithPrimaryKey>().count().find())
+            assertEquals(10, query<EmbeddedParent>().count().find())
+            assertEquals(10, query<EmbeddedChild>().count().find())
```

Comment:
These asserts don't need to be inside a `writeBlocking`. They should work fine when moved outside and then using `realm.query()`.

Reply:
You can also do this `with(realm) { asssertX }` or `realm.run { assertX }`

Reply:
We don't have any strict guidelines about which to use, but I would probably consider `with(realm) { }` the most readable